{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "gpuType": "T4",
      "authorship_tag": "ABX9TyO5kN1T7uCeotQeiI47Pf+O",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "accelerator": "GPU"
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/renatocueva/Renatocueva/blob/main/Percepcion3D.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3WtwWcMlKSg7",
        "outputId": "c3bf05bd-d26b-425e-de3b-a6598fc8a3b2"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Found GPU at: /device:GPU:0\n"
          ]
        }
      ],
      "source": [
        "import tensorflow as tf\n",
        "device_name=tf.test.gpu_device_name()\n",
        "if device_name !='/device:GPU:0':\n",
        "    raise SystemError('GPU device not found')\n",
        "print('Found GPU at: {}'.format(device_name))"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "m0N-ZCVDKWDy",
        "outputId": "40e6854b-2b8e-4bfa-8086-0047242afffd"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget -N https://github.com/alicevision/meshroom/releases/download/v2019.1.0/Meshroom-2019.1.0-linux.tar.gz\n",
        "!mkdir meshroom\n",
        "!tar xzf Meshroom-2019.1.0-linux.tar.gz -C ./meshroom"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "34i2B5l7KbjT",
        "outputId": "7126062a-a4bb-4375-ed3e-0d3dddb07890"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2024-05-26 10:32:37--  https://github.com/alicevision/meshroom/releases/download/v2019.1.0/Meshroom-2019.1.0-linux.tar.gz\n",
            "Resolving github.com (github.com)... 140.82.116.4\n",
            "Connecting to github.com (github.com)|140.82.116.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/34405381/ac2a6000-44ad-11e9-9c7e-7405269e659a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240526%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240526T103238Z&X-Amz-Expires=300&X-Amz-Signature=cdf9bc8f07fa77359f6afdd6d036b0de922193e8623db792bac0a908d1dcda77&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=34405381&response-content-disposition=attachment%3B%20filename%3DMeshroom-2019.1.0-linux.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
            "--2024-05-26 10:32:38--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/34405381/ac2a6000-44ad-11e9-9c7e-7405269e659a?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20240526%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240526T103238Z&X-Amz-Expires=300&X-Amz-Signature=cdf9bc8f07fa77359f6afdd6d036b0de922193e8623db792bac0a908d1dcda77&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=34405381&response-content-disposition=attachment%3B%20filename%3DMeshroom-2019.1.0-linux.tar.gz&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 304 Not Modified\n",
            "File ‘Meshroom-2019.1.0-linux.tar.gz’ not modified on server. Omitting download.\n",
            "\n",
            "mkdir: cannot create directory ‘meshroom’: File exists\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir ./object_out\n",
        "!./meshroom/Meshroom-2019.1.0/meshroom_photogrammetry --input /content/drive/MyDrive/FotosBotella --output ./object_out"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "h-azrareLgnM",
        "outputId": "1c335639-bbde-401c-b93c-8e584d1535d8"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "mkdir: cannot create directory ‘./object_out’: File exists\n",
            "Plugins loaded:  CameraCalibration, CameraInit, CameraLocalization, CameraRigCalibration, CameraRigLocalization, ConvertSfMFormat, DepthMap, DepthMapFilter, ExportAnimatedCamera, ExportMaya, FeatureExtraction, FeatureMatching, ImageMatching, ImageMatchingMultiSfM, KeyframeSelection, MeshDecimate, MeshDenoising, MeshFiltering, MeshResampling, Meshing, PrepareDenseScene, Publish, SfMAlignment, SfMTransform, StructureFromMotion, Texturing\n",
            "Program called with the following parameters:\n",
            " * allowSingleView = 1\n",
            " * defaultCameraModel = \"\" (default)\n",
            " * defaultFieldOfView = 45\n",
            " * defaultFocalLengthPix = -1 (default)\n",
            " * defaultIntrinsic = \"\" (default)\n",
            " * groupCameraFallback =  Unknown Type \"20EGroupCameraFallback\"\n",
            " * imageFolder = \"\" (default)\n",
            " * input = \"/tmp/tmpzluwf2wh/CameraInit/c448939571d5c70b05c9ae4ad416ada37d6273ca//viewpoints.sfm\"\n",
            " * output = \"/tmp/tmpzluwf2wh/CameraInit/c448939571d5c70b05c9ae4ad416ada37d6273ca/cameraInit.sfm\"\n",
            " * sensorDatabase = \"/content/meshroom/Meshroom-2019.1.0/aliceVision/share/aliceVision/cameraSensors.db\"\n",
            " * verboseLevel = \"info\"\n",
            "\n",
            "[10:37:09.272091][warning] Image 'WhatsApp Image 2024-05-26 at 11.17.57.jpg' focal length (in mm) metadata is missing.\n",
            "Can't compute focal length (px), use default.\n",
            "\n",
            "[10:37:09.272073][warning] Image 'WhatsApp Image 2024-05-26 at 11.17.53.jpg' focal length (in mm) metadata is missing.\n",
            "Can't compute focal length (px), use default.\n",
            "\n",
            "[10:37:09.272241][warning] Image 'WhatsApp Image 2024-05-26 at 11.17.52 (1).jpg' focal length (in mm) metadata is missing.\n",
            "Can't compute focal length (px), use default.\n",
            "\n",
            "[10:37:09.272241][warning] Image 'WhatsApp Image 2024-05-26 at 11.17.58 (1).jpg' focal length (in mm) metadata is missing.\n",
            "Can't compute focal length (px), use default.\n",
            "\n",
            "[10:37:09.272276][warning] Image 'WhatsApp Image 2024-05-26 at 11.17.57 (1).jpg' focal length (in mm) metadata is missing.\n",
            "Can't compute focal length (px), use default.\n",
            "\n",
            "[10:37:09.272295][warning] Image 'WhatsApp Image 2024-05-26 at 11.17.52.jpg' focal length (in mm) metadata is missing.\n",
            "Can't compute focal length (px), use default.\n",
            "\n",
            "[10:37:09.272299][warning] Image 'WhatsApp Image 2024-05-26 at 11.17.56.jpg' focal length (in mm) metadata is missing.\n",
            "Can't compute focal length (px), use default.\n",
            "\n",
            "[10:37:09.272329][warning] Image 'WhatsApp Image 2024-05-26 at 11.17.54 (1).jpg' focal length (in mm) metadata is missing.\n",
            "Can't compute focal length (px), use default.\n",
            "\n",
            "[10:37:09.272345][warning] Image 'WhatsApp Image 2024-05-26 at 11.17.58.jpg' focal length (in mm) metadata is missing.\n",
            "Can't compute focal length (px), use default.\n",
            "\n",
            "[10:37:09.272353][warning] Image 'WhatsApp Image 2024-05-26 at 11.17.55 (1).jpg' focal length (in mm) metadata is missing.\n",
            "Can't compute focal length (px), use default.\n",
            "\n",
            "[10:37:09.272374][warning] Image 'WhatsApp Image 2024-05-26 at 11.17.55.jpg' focal length (in mm) metadata is missing.\n",
            "Can't compute focal length (px), use default.\n",
            "\n",
            "[10:37:09.272382][warning] Image 'WhatsApp Image 2024-05-26 at 11.17.59.jpg' focal length (in mm) metadata is missing.\n",
            "Can't compute focal length (px), use default.\n",
            "\n",
            "[10:37:09.272404][warning] Image 'WhatsApp Image 2024-05-26 at 11.17.56 (1).jpg' focal length (in mm) metadata is missing.\n",
            "Can't compute focal length (px), use default.\n",
            "\n",
            "[10:37:09.272410][warning] Image 'WhatsApp Image 2024-05-26 at 11.17.54.jpg' focal length (in mm) metadata is missing.\n",
            "Can't compute focal length (px), use default.\n",
            "\n",
            "[10:37:09.273033][warning] Some image(s) have no serial number to identify the camera/lens device.\n",
            "This makes it impossible to correctly group the images by device if you have used multiple identical (same model) camera devices.\n",
            "The reconstruction will assume that only one device has been used, so if 2 images share the same focal length approximation they will share the same internal camera parameters.\n",
            "14 image(s) are concerned.\n",
            "[10:37:09.274123][info] CameraInit report:\n",
            "\t- # views listed: 14\n",
            "\t   - # views with an initialized intrinsic listed: 14\n",
            "\t   - # views without metadata (with a default intrinsic): 14\n",
            "\t- # intrinsics listed: 1\n",
            "Nodes to execute:  ['CameraInit_1', 'FeatureExtraction_1', 'ImageMatching_1', 'FeatureMatching_1', 'StructureFromMotion_1', 'PrepareDenseScene_1', 'DepthMap_1', 'DepthMapFilter_1', 'Meshing_1', 'MeshFiltering_1', 'Texturing_1', 'Publish_1']\n",
            "\n",
            "[1/12] CameraInit\n",
            " - commandLine: aliceVision_cameraInit  --sensorDatabase \"/content/meshroom/Meshroom-2019.1.0/aliceVision/share/aliceVision/cameraSensors.db\" --defaultFieldOfView 45.0 --groupCameraFallback folder --verboseLevel info --output \"/tmp/MeshroomCache/CameraInit/5b2eebc332b67dcd0a34f8cecbdf2e7510e1752d/cameraInit.sfm\" --allowSingleView 1  --input \"/tmp/MeshroomCache/CameraInit/5b2eebc332b67dcd0a34f8cecbdf2e7510e1752d//viewpoints.sfm\"\n",
            " - logFile: /tmp/MeshroomCache/CameraInit/5b2eebc332b67dcd0a34f8cecbdf2e7510e1752d/log\n",
            "Exception in thread Thread-1:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71017 0 5489992 68882 0 47567 294360 0 0 0 0 1116 6034\\n'\n",
            "\n",
            " - elapsed time: 0:00:00.036223\n",
            "\n",
            "[2/12] FeatureExtraction\n",
            " - commandLine: aliceVision_featureExtraction  --input \"/tmp/MeshroomCache/CameraInit/5b2eebc332b67dcd0a34f8cecbdf2e7510e1752d/cameraInit.sfm\" --describerTypes sift --describerPreset normal --forceCpuExtraction True --verboseLevel info --output \"/tmp/MeshroomCache/FeatureExtraction/15b80157a86af676d0b489279db5617f2aefac78\" --rangeStart 0 --rangeSize 15\n",
            " - logFile: /tmp/MeshroomCache/FeatureExtraction/15b80157a86af676d0b489279db5617f2aefac78/0.log\n",
            "Exception in thread Thread-2:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71020 0 5490016 68882 0 47570 294360 0 0 0 0 1116 6034\\n'\n",
            "\n",
            " - elapsed time: 0:00:12.848304\n",
            "\n",
            "[3/12] ImageMatching\n",
            " - commandLine: aliceVision_imageMatching  --input \"/tmp/MeshroomCache/CameraInit/5b2eebc332b67dcd0a34f8cecbdf2e7510e1752d/cameraInit.sfm\" --featuresFolders \"/tmp/MeshroomCache/FeatureExtraction/15b80157a86af676d0b489279db5617f2aefac78\" --tree \"/content/meshroom/Meshroom-2019.1.0/aliceVision/share/aliceVision/vlfeat_K80L3.SIFT.tree\" --weights \"\" --minNbImages 200 --maxDescriptors 500 --nbMatches 50 --verboseLevel info --output \"/tmp/MeshroomCache/ImageMatching/089ce4ec1213047371a26c27bcbe4467989aecb6/imageMatches.txt\" \n",
            " - logFile: /tmp/MeshroomCache/ImageMatching/089ce4ec1213047371a26c27bcbe4467989aecb6/log\n",
            "Exception in thread Thread-3:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71135 0 5493184 68932 0 47596 294427 0 0 0 0 1126 6052\\n'\n",
            "\n",
            " - elapsed time: 0:00:00.023361\n",
            "\n",
            "[4/12] FeatureMatching\n",
            " - commandLine: aliceVision_featureMatching  --input \"/tmp/MeshroomCache/CameraInit/5b2eebc332b67dcd0a34f8cecbdf2e7510e1752d/cameraInit.sfm\" --featuresFolders \"/tmp/MeshroomCache/FeatureExtraction/15b80157a86af676d0b489279db5617f2aefac78\" --imagePairsList \"/tmp/MeshroomCache/ImageMatching/089ce4ec1213047371a26c27bcbe4467989aecb6/imageMatches.txt\" --describerTypes sift --photometricMatchingMethod ANN_L2 --geometricEstimator acransac --geometricFilterType fundamental_matrix --distanceRatio 0.8 --maxIteration 2048 --geometricError 0.0 --maxMatches 0 --savePutativeMatches False --guidedMatching False --exportDebugFiles False --verboseLevel info --output \"/tmp/MeshroomCache/FeatureMatching/f4b8faf691825da20f7d9522bf48163559d05443\" --rangeStart 0 --rangeSize 15\n",
            " - logFile: /tmp/MeshroomCache/FeatureMatching/f4b8faf691825da20f7d9522bf48163559d05443/0.log\n",
            "Exception in thread Thread-4:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71138 0 5493208 68932 0 47598 294427 0 0 0 0 1126 6052\\n'\n",
            "\n",
            " - elapsed time: 0:00:17.898180\n",
            "\n",
            "[5/12] StructureFromMotion\n",
            " - commandLine: aliceVision_incrementalSfM  --input \"/tmp/MeshroomCache/CameraInit/5b2eebc332b67dcd0a34f8cecbdf2e7510e1752d/cameraInit.sfm\" --featuresFolders \"/tmp/MeshroomCache/FeatureExtraction/15b80157a86af676d0b489279db5617f2aefac78\" --matchesFolders \"/tmp/MeshroomCache/FeatureMatching/f4b8faf691825da20f7d9522bf48163559d05443\" --describerTypes sift --localizerEstimator acransac --localizerEstimatorMaxIterations 4096 --localizerEstimatorError 0.0 --lockScenePreviouslyReconstructed False --useLocalBA True --localBAGraphDistance 1 --maxNumberOfMatches 0 --minInputTrackLength 2 --minNumberOfObservationsForTriangulation 2 --minAngleForTriangulation 3.0 --minAngleForLandmark 2.0 --maxReprojectionError 4.0 --minAngleInitialPair 5.0 --maxAngleInitialPair 40.0 --useOnlyMatchesFromInputFolder False --useRigConstraint True --lockAllIntrinsics False --initialPairA \"\" --initialPairB \"\" --interFileExtension .abc --verboseLevel info --output \"/tmp/MeshroomCache/StructureFromMotion/6305c1f0e0241b3cce82742bbfb89b07440a321a/sfm.abc\" --outputViewsAndPoses \"/tmp/MeshroomCache/StructureFromMotion/6305c1f0e0241b3cce82742bbfb89b07440a321a/cameras.sfm\" --extraInfoFolder \"/tmp/MeshroomCache/StructureFromMotion/6305c1f0e0241b3cce82742bbfb89b07440a321a\" \n",
            " - logFile: /tmp/MeshroomCache/StructureFromMotion/6305c1f0e0241b3cce82742bbfb89b07440a321a/log\n",
            "Exception in thread Thread-5:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71271 0 5494368 68959 0 47620 294469 0 0 0 0 1132 6066\\n'\n",
            "\n",
            " - elapsed time: 0:00:01.327400\n",
            "\n",
            "[6/12] PrepareDenseScene\n",
            " - commandLine: aliceVision_prepareDenseScene  --input \"/tmp/MeshroomCache/StructureFromMotion/6305c1f0e0241b3cce82742bbfb89b07440a321a/sfm.abc\" --outputFileType exr --saveMetadata True --saveMatricesTxtFiles False --verboseLevel info --output \"/tmp/MeshroomCache/PrepareDenseScene/7ed6a3ad1b7216e1915813726f36e718ccc0c84f\" --rangeStart 0 --rangeSize 15\n",
            " - logFile: /tmp/MeshroomCache/PrepareDenseScene/7ed6a3ad1b7216e1915813726f36e718ccc0c84f/0.log\n",
            "Exception in thread Thread-6:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71291 0 5494528 68965 0 47627 294479 0 0 0 0 1134 6070\\n'\n",
            "\n",
            " - elapsed time: 0:00:00.979089\n",
            "\n",
            "[7/12](1/5) DepthMap\n",
            " - commandLine: aliceVision_depthMapEstimation  --input \"/tmp/MeshroomCache/StructureFromMotion/6305c1f0e0241b3cce82742bbfb89b07440a321a/sfm.abc\" --imagesFolder \"/tmp/MeshroomCache/PrepareDenseScene/7ed6a3ad1b7216e1915813726f36e718ccc0c84f\" --downscale 2 --minViewAngle 2.0 --maxViewAngle 70.0 --sgmMaxTCams 10 --sgmWSH 4 --sgmGammaC 5.5 --sgmGammaP 8.0 --refineMaxTCams 6 --refineNSamplesHalf 150 --refineNDepthsToRefine 31 --refineNiters 100 --refineWSH 3 --refineSigma 15 --refineGammaC 15.5 --refineGammaP 8.0 --refineUseTcOrRcPixSize False --exportIntermediateResults False --nbGPUs 0 --verboseLevel info --output \"/tmp/MeshroomCache/DepthMap/dd32947dc4b025c3d90b9945471f84c36fa9e7d0\" --rangeStart 0 --rangeSize 3\n",
            " - logFile: /tmp/MeshroomCache/DepthMap/dd32947dc4b025c3d90b9945471f84c36fa9e7d0/0.log\n",
            "Exception in thread Thread-7:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71294 0 5494552 68965 0 47629 294479 0 0 0 0 1134 6070\\n'\n",
            "\n",
            " - elapsed time: 0:00:06.240103\n",
            "\n",
            "[7/12](2/5) DepthMap\n",
            " - commandLine: aliceVision_depthMapEstimation  --input \"/tmp/MeshroomCache/StructureFromMotion/6305c1f0e0241b3cce82742bbfb89b07440a321a/sfm.abc\" --imagesFolder \"/tmp/MeshroomCache/PrepareDenseScene/7ed6a3ad1b7216e1915813726f36e718ccc0c84f\" --downscale 2 --minViewAngle 2.0 --maxViewAngle 70.0 --sgmMaxTCams 10 --sgmWSH 4 --sgmGammaC 5.5 --sgmGammaP 8.0 --refineMaxTCams 6 --refineNSamplesHalf 150 --refineNDepthsToRefine 31 --refineNiters 100 --refineWSH 3 --refineSigma 15 --refineGammaC 15.5 --refineGammaP 8.0 --refineUseTcOrRcPixSize False --exportIntermediateResults False --nbGPUs 0 --verboseLevel info --output \"/tmp/MeshroomCache/DepthMap/dd32947dc4b025c3d90b9945471f84c36fa9e7d0\" --rangeStart 3 --rangeSize 3\n",
            " - logFile: /tmp/MeshroomCache/DepthMap/dd32947dc4b025c3d90b9945471f84c36fa9e7d0/1.log\n",
            "Exception in thread Thread-8:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71369 0 5509960 69069 0 47653 294596 0 0 0 0 1136 6084\\n'\n",
            "\n",
            " - elapsed time: 0:00:00.322124\n",
            "\n",
            "[7/12](3/5) DepthMap\n",
            " - commandLine: aliceVision_depthMapEstimation  --input \"/tmp/MeshroomCache/StructureFromMotion/6305c1f0e0241b3cce82742bbfb89b07440a321a/sfm.abc\" --imagesFolder \"/tmp/MeshroomCache/PrepareDenseScene/7ed6a3ad1b7216e1915813726f36e718ccc0c84f\" --downscale 2 --minViewAngle 2.0 --maxViewAngle 70.0 --sgmMaxTCams 10 --sgmWSH 4 --sgmGammaC 5.5 --sgmGammaP 8.0 --refineMaxTCams 6 --refineNSamplesHalf 150 --refineNDepthsToRefine 31 --refineNiters 100 --refineWSH 3 --refineSigma 15 --refineGammaC 15.5 --refineGammaP 8.0 --refineUseTcOrRcPixSize False --exportIntermediateResults False --nbGPUs 0 --verboseLevel info --output \"/tmp/MeshroomCache/DepthMap/dd32947dc4b025c3d90b9945471f84c36fa9e7d0\" --rangeStart 6 --rangeSize 3\n",
            " - logFile: /tmp/MeshroomCache/DepthMap/dd32947dc4b025c3d90b9945471f84c36fa9e7d0/2.log\n",
            "Exception in thread Thread-9:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71372 0 5509984 69069 0 47655 294597 0 0 0 0 1136 6084\\n'\n",
            "\n",
            " - elapsed time: 0:00:00.301598\n",
            "\n",
            "[7/12](4/5) DepthMap\n",
            " - commandLine: aliceVision_depthMapEstimation  --input \"/tmp/MeshroomCache/StructureFromMotion/6305c1f0e0241b3cce82742bbfb89b07440a321a/sfm.abc\" --imagesFolder \"/tmp/MeshroomCache/PrepareDenseScene/7ed6a3ad1b7216e1915813726f36e718ccc0c84f\" --downscale 2 --minViewAngle 2.0 --maxViewAngle 70.0 --sgmMaxTCams 10 --sgmWSH 4 --sgmGammaC 5.5 --sgmGammaP 8.0 --refineMaxTCams 6 --refineNSamplesHalf 150 --refineNDepthsToRefine 31 --refineNiters 100 --refineWSH 3 --refineSigma 15 --refineGammaC 15.5 --refineGammaP 8.0 --refineUseTcOrRcPixSize False --exportIntermediateResults False --nbGPUs 0 --verboseLevel info --output \"/tmp/MeshroomCache/DepthMap/dd32947dc4b025c3d90b9945471f84c36fa9e7d0\" --rangeStart 9 --rangeSize 3\n",
            " - logFile: /tmp/MeshroomCache/DepthMap/dd32947dc4b025c3d90b9945471f84c36fa9e7d0/3.log\n",
            "Exception in thread Thread-10:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71375 0 5510008 69071 0 47658 294598 0 0 0 0 1136 6084\\n'\n",
            "\n",
            " - elapsed time: 0:00:00.313231\n",
            "\n",
            "[7/12](5/5) DepthMap\n",
            " - commandLine: aliceVision_depthMapEstimation  --input \"/tmp/MeshroomCache/StructureFromMotion/6305c1f0e0241b3cce82742bbfb89b07440a321a/sfm.abc\" --imagesFolder \"/tmp/MeshroomCache/PrepareDenseScene/7ed6a3ad1b7216e1915813726f36e718ccc0c84f\" --downscale 2 --minViewAngle 2.0 --maxViewAngle 70.0 --sgmMaxTCams 10 --sgmWSH 4 --sgmGammaC 5.5 --sgmGammaP 8.0 --refineMaxTCams 6 --refineNSamplesHalf 150 --refineNDepthsToRefine 31 --refineNiters 100 --refineWSH 3 --refineSigma 15 --refineGammaC 15.5 --refineGammaP 8.0 --refineUseTcOrRcPixSize False --exportIntermediateResults False --nbGPUs 0 --verboseLevel info --output \"/tmp/MeshroomCache/DepthMap/dd32947dc4b025c3d90b9945471f84c36fa9e7d0\" --rangeStart 12 --rangeSize 3\n",
            " - logFile: /tmp/MeshroomCache/DepthMap/dd32947dc4b025c3d90b9945471f84c36fa9e7d0/4.log\n",
            "Exception in thread Thread-11:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71378 0 5510032 69071 0 47661 294598 0 0 0 0 1136 6084\\n'\n",
            "\n",
            " - elapsed time: 0:00:00.342093\n",
            "\n",
            "[8/12](1/2) DepthMapFilter\n",
            " - commandLine: aliceVision_depthMapFiltering  --input \"/tmp/MeshroomCache/StructureFromMotion/6305c1f0e0241b3cce82742bbfb89b07440a321a/sfm.abc\" --depthMapsFolder \"/tmp/MeshroomCache/DepthMap/dd32947dc4b025c3d90b9945471f84c36fa9e7d0\" --minViewAngle 2.0 --maxViewAngle 70.0 --nNearestCams 10 --minNumOfConsistentCams 3 --minNumOfConsistentCamsWithLowSimilarity 4 --pixSizeBall 0 --pixSizeBallWithLowSimilarity 0 --verboseLevel info --output \"/tmp/MeshroomCache/DepthMapFilter/36fd6b709d6a33e61f31fd8fac6fbbfabb6c94f9\" --rangeStart 0 --rangeSize 10\n",
            " - logFile: /tmp/MeshroomCache/DepthMapFilter/36fd6b709d6a33e61f31fd8fac6fbbfabb6c94f9/0.log\n",
            "Exception in thread Thread-12:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71381 0 5510056 69071 0 47664 294598 0 0 0 0 1136 6084\\n'\n",
            "\n",
            " - elapsed time: 0:00:02.156314\n",
            "\n",
            "[8/12](2/2) DepthMapFilter\n",
            " - commandLine: aliceVision_depthMapFiltering  --input \"/tmp/MeshroomCache/StructureFromMotion/6305c1f0e0241b3cce82742bbfb89b07440a321a/sfm.abc\" --depthMapsFolder \"/tmp/MeshroomCache/DepthMap/dd32947dc4b025c3d90b9945471f84c36fa9e7d0\" --minViewAngle 2.0 --maxViewAngle 70.0 --nNearestCams 10 --minNumOfConsistentCams 3 --minNumOfConsistentCamsWithLowSimilarity 4 --pixSizeBall 0 --pixSizeBallWithLowSimilarity 0 --verboseLevel info --output \"/tmp/MeshroomCache/DepthMapFilter/36fd6b709d6a33e61f31fd8fac6fbbfabb6c94f9\" --rangeStart 10 --rangeSize 5\n",
            " - logFile: /tmp/MeshroomCache/DepthMapFilter/36fd6b709d6a33e61f31fd8fac6fbbfabb6c94f9/1.log\n",
            "Exception in thread Thread-13:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71412 0 5510288 69086 0 47681 294626 0 0 0 0 1140 6096\\n'\n",
            "\n",
            " - elapsed time: 0:00:00.029895\n",
            "\n",
            "[9/12] Meshing\n",
            " - commandLine: aliceVision_meshing  --input \"/tmp/MeshroomCache/StructureFromMotion/6305c1f0e0241b3cce82742bbfb89b07440a321a/sfm.abc\" --depthMapsFolder \"/tmp/MeshroomCache/DepthMap/dd32947dc4b025c3d90b9945471f84c36fa9e7d0\" --depthMapsFilterFolder \"/tmp/MeshroomCache/DepthMapFilter/36fd6b709d6a33e61f31fd8fac6fbbfabb6c94f9\" --estimateSpaceFromSfM True --estimateSpaceMinObservations 3 --estimateSpaceMinObservationAngle 10 --maxInputPoints 50000000 --maxPoints 5000000 --maxPointsPerVoxel 1000000 --minStep 2 --partitioning singleBlock --repartition multiResolution --angleFactor 15.0 --simFactor 15.0 --pixSizeMarginInitCoef 2.0 --pixSizeMarginFinalCoef 4.0 --voteMarginFactor 4.0 --contributeMarginFactor 2.0 --simGaussianSizeInit 10.0 --simGaussianSize 10.0 --minAngleThreshold 1.0 --refineFuse True --addLandmarksToTheDensePointCloud False --verboseLevel info --output \"/tmp/MeshroomCache/Meshing/ff92e53ff89dc0d4f6e3e63b4b46fcee6f634285/mesh.obj\" \n",
            " - logFile: /tmp/MeshroomCache/Meshing/ff92e53ff89dc0d4f6e3e63b4b46fcee6f634285/log\n",
            "Exception in thread Thread-14:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71415 0 5510312 69087 0 47684 294626 0 0 0 0 1140 6096\\n'\n",
            "\n",
            " - elapsed time: 0:00:05.336503\n",
            "\n",
            "[10/12] MeshFiltering\n",
            " - commandLine: aliceVision_meshFiltering  --input \"/tmp/MeshroomCache/Meshing/ff92e53ff89dc0d4f6e3e63b4b46fcee6f634285/mesh.obj\" --removeLargeTrianglesFactor 60.0 --keepLargestMeshOnly False --iterations 5 --lambda 1.0 --verboseLevel info --output \"/tmp/MeshroomCache/MeshFiltering/0327a4bd82e5fdc6bbfc7d18ff3675945ada1954/mesh.obj\" \n",
            " - logFile: /tmp/MeshroomCache/MeshFiltering/0327a4bd82e5fdc6bbfc7d18ff3675945ada1954/log\n",
            "Exception in thread Thread-15:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71452 0 5518072 69115 0 47702 294664 0 0 0 0 1142 6105\\n'\n",
            "\n",
            " - elapsed time: 0:00:00.201063\n",
            "\n",
            "[11/12] Texturing\n",
            " - commandLine: aliceVision_texturing  --input \"/tmp/MeshroomCache/StructureFromMotion/6305c1f0e0241b3cce82742bbfb89b07440a321a/sfm.abc\" --imagesFolder \"/tmp/MeshroomCache/PrepareDenseScene/7ed6a3ad1b7216e1915813726f36e718ccc0c84f\" --inputDenseReconstruction \"/tmp/MeshroomCache/Meshing/ff92e53ff89dc0d4f6e3e63b4b46fcee6f634285/denseReconstruction.bin\" --inputMesh \"/tmp/MeshroomCache/MeshFiltering/0327a4bd82e5fdc6bbfc7d18ff3675945ada1954/mesh.obj\" --textureSide 8192 --downscale 2 --outputTextureFileType png --unwrapMethod Basic --fillHoles False --padding 15 --maxNbImagesForFusion 3 --bestScoreThreshold 0.0 --angleHardThreshold 90.0 --forceVisibleByAllVertices False --flipNormals False --visibilityRemappingMethod PullPush --verboseLevel info --output \"/tmp/MeshroomCache/Texturing/b418120049311e30295191086d7d4ae6b43cf38b\" \n",
            " - logFile: /tmp/MeshroomCache/Texturing/b418120049311e30295191086d7d4ae6b43cf38b/log\n",
            "Exception in thread Thread-16:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71455 0 5518096 69115 0 47705 294664 0 0 0 0 1142 6105\\n'\n",
            "\n",
            " - elapsed time: 0:00:25.244893\n",
            "\n",
            "[12/12] Publish\n",
            "Publish\n",
            "Publish file /tmp/MeshroomCache/Texturing/b418120049311e30295191086d7d4ae6b43cf38b/texturedMesh.obj into object_out/texturedMesh.obj\n",
            "Exception in thread Thread-17:\n",
            "Traceback (most recent call last):\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 206, in run\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 200, in updateStats\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 162, in update\n",
            "  File \"/opt/Meshroom/meshroom/core/stats.py\", line 52, in update\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/__init__.py\", line 2131, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1121, in disk_io_counters\n",
            "  File \"/opt/rh/rh-python36/root/usr/lib64/python3.6/site-packages/psutil/_pslinux.py\", line 1094, in read_procfs\n",
            "ValueError: not sure how to interpret line '   7       0 loop0 107243 0 3833314 219443 71629 0 5583408 69861 0 47798 295455 0 0 0 0 1151 6150\\n'\n",
            "\n",
            "Publish file /tmp/MeshroomCache/Texturing/b418120049311e30295191086d7d4ae6b43cf38b/texturedMesh.mtl into object_out/texturedMesh.mtl\n",
            "Publish file /tmp/MeshroomCache/Texturing/b418120049311e30295191086d7d4ae6b43cf38b/texture_0.png into object_out/texture_0.png\n",
            "Publish end\n",
            " - elapsed time: 0:00:00.003973\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from google.colab import files\n",
        "\n",
        "!zip -r meshobject.zip ./object_out\n",
        "files.download('meshobject.zip')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 86
        },
        "id": "TX4TjdqnLpYH",
        "outputId": "f1c8454a-ee3b-4c28-fc9f-981a6589b44f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "updating: object_out/ (stored 0%)\n",
            "  adding: object_out/texturedMesh.obj (deflated 66%)\n",
            "  adding: object_out/texturedMesh.mtl (deflated 28%)\n",
            "  adding: object_out/texture_0.png (deflated 11%)\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_88c438c6-a9a6-40f4-b086-b2bbcb892d58\", \"meshobject.zip\", 553228)"
            ]
          },
          "metadata": {}
        }
      ]
    }
  ]
}